subiquity/models/source.py

```diff
@@ -89,14 +89,19 @@ def load_from_file(self, fp):
 
     def get_source(self):
         path = os.path.join(self._dir, self.current.path)
+        scheme = self.current.type
+
+        if self.current.path.startswith("http"):
```

Comment:
This would misbehave if the local file name really starts with http.  This is arguably a poor choice for a file name but I can see it happening if someone caches a local copy of the file and uses the URI as the input for generation of the name.  So I would like a deeper inspection of the path value to check if it looks like a URI or not.